presto-main/src/main/java/io/prestosql/connector/system/KillQueryProcedure.java

```diff
@@ -43,11 +45,13 @@
     private static final MethodHandle KILL_QUERY = methodHandle(KillQueryProcedure.class, "killQuery", String.class, String.class);
 
     private final QueryManager queryManager;
+    private final Optional<DispatchManager> dispatchManager;
 
     @Inject
-    public KillQueryProcedure(QueryManager queryManager)
+    public KillQueryProcedure(QueryManager queryManager, Optional<DispatchManager> dispatchManager)
```

Comment:
Only bind this in the coordinator and then you don't need the optional

Reply:
Then I would need to break `SystemConnectorModule` into several modules. Not sure what is better. I would refrain from this refactactor in this PR. Let me fix this as a follow up